### PR TITLE
Added Sorting and Paging Example and some helpful Query constructors

### PR DESCRIPTION
**Pull request by cosbor11**

### CODE REVIEW

File: onyx-database-examples/querying/src/main/java/com/onyxdevtools/example/querying/LazyQueryExample.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        final PersistenceManager manager = factory.getPersistenceManager();
+
+        // Create a query and criteria
+        final QueryCriteria criteria = new QueryCriteria("position", QueryCriteriaOperator.EQUAL, "QB");
+        criteria.and("lastName", QueryCriteriaOperator.STARTS_WITH, "C");
+
+        final Query query = new Query(Player.class, criteria, new QueryOrder("firstName"));
+        // query.setCriteria(criteria); or you can set the critiera after construction
+
+        final List<Player> quarterbacks = manager.executeQuery(query);
```
Comment (timbob1000) [inline onyx-database-examples/querying/src/main/java/com/onyxdevtools/example/querying/LazyQueryExample.java]: This doesn't actually use the executeLazyQuery API.

File: onyx-database-examples/querying/src/main/java/com/onyxdevtools/example/querying/LazyQueryExample.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        for (final Player qb : quarterbacks)
```
Comment (timbob1000) [inline onyx-database-examples/querying/src/main/java/com/onyxdevtools/example/querying/LazyQueryExample.java]: Not sure if fast enumeration will work.  I suppose this is a good test

File: onyx-database-examples/querying/src/main/java/com/onyxdevtools/example/querying/LazyQueryExample.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import java.io.File;
+import java.io.IOException;
+
+import java.util.List;
+
+
+/**
+ @author  cosborn
+ */
+public class LazyQueryExample
```
Comment (timbob1000) [inline onyx-database-examples/querying/src/main/java/com/onyxdevtools/example/querying/LazyQueryExample.java]: Can you add some comments to this class that addresses the benefit of what the lazy query has.  In the event a user just downloads the example project and doesn't go through the tutorial.

File: onyx-database-examples/querying/src/main/java/com/onyxdevtools/example/querying/LazyQueryExample.java (hunk 1 above)
Comment (cosbor11) [inline onyx-database-examples/querying/src/main/java/com/onyxdevtools/example/querying/LazyQueryExample.java]: I just added the LazyQuery as a placeholder. I'll stub the rest out later today

**Pull request merged**